specification/communication/data-plane/JobRouter/preview/2021-10-20-preview2/communicationservicejobrouter.json

```diff
@@ -0,0 +1,3635 @@
+          "application/merge-patch+json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "Id of the job",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "forceClassification",
+            "description": "If set to true, will force classification. Defaults to false.",
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "patch",
+            "description": "Request model for patching a job",
+            "schema": {
+              "$ref": "#/definitions/RouterJob"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/RouterJob"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update a job": {
+            "$ref": "./examples/Jobs_UpdateJob.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Jobs"
+        ],
+        "summary": "Deletes a job and all of its traces.",
+        "operationId": "JobRouter_DeleteJob",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Success"
+          }
+        },
+        "x-ms-examples": {
+          "Deletes a job and all of its traces": {
+            "$ref": "./examples/Jobs_DeleteJob.json"
+          }
+        }
+      }
+    },
+    "/routing/jobs/{id}:cancel": {
+      "post": {
+        "tags": [
+          "Jobs"
+        ],
+        "summary": "Submits request to cancel an existing job by Id while supplying free-form cancellation reason.",
+        "operationId": "JobRouter_CancelJobAction",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "Id of the job",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "cancelJobRequest",
+            "description": "Request model for cancelling job",
+            "schema": {
+              "$ref": "#/definitions/CancelJobRequest"
+            },
+            "x-ms-client-flatten": true
```

Comment:
Avoid using `x-ms-client-flatten` -- it is problematic for SDK Gen tooling.

Reply:
I believe that the usage here is correct according to https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md?msclkid=c91a58b2a17311ecbc9d3910e194499b#x-ms-client-flatten

Reply:
Our Data Plane Codegen generator, which will be used as the basis of all SDK generation going forward, ignores this extension. I will open an issue to update the documentation.

Reply:
Perhaps I made some assumptions about how you would be generating your client side libraries that aren't valid. So let's keep this in place for now but pls be aware that the newer generators will likely not support this.